commands.go

```diff
@@ -86,3 +87,31 @@ func decryptEnvironment(input []string, output io.Writer, crypto DecryptionStrat
 
 	return ok, err
 }
+
+func createExecArgs(args []string, encryptedEnviron []string, crypto DecryptionStrategy) (cmd string, decryptedArgs []string, decryptedEnviron []string, err error) {
+
+	cmd = args[0]
+	decryptedArgs = make([]string, len(args))
+
+	decryptedArgs[0] = path.Base(cmd) // By unix convention argv[0] has to be set to basename of command
+	for i, arg := range args[1:] {
+		decryptedArg, subErr := decryptEnvelopes(arg, crypto)
```

Comment:
Could you consider refactoring atleast decryptStream() and possibly decryptEnvironment() to use the decryptEnvelopes() function as well? Cut's down the complexity now that you've extracted a nice reusable function here